lib/adapters/parsers/grok.js

```diff
@@ -0,0 +1,71 @@
+var base = require('./base');
+var errors = require('../../errors');
+var grok = require('node-grok');
+var Promise = require('bluebird');
+var readline = require('readline');
+var sync = require('sync');
+
+module.exports = base.extend({
+
+    initialize: function(options) {
+        this.pattern = options.pattern || '%{GREEDYDATA:message}';
+    },
+
+    parseStream: function(stream, emit) {
+        var self = this;
+
+        return new Promise(function(resolve, reject) {
+            var buffer = [];
+            var pending = 0;
+            var streamEnd = false;
+
+            function checkDone(streamEnd) {
+                if (buffer.length >= self.limit) {
+                    emit(buffer);
+                    buffer = [];
+                }
+
+                if (pending === 0 && streamEnd) {
+                    emit(buffer);
+                    buffer = [];
+                    resolve();
+                }
+            }
+
+            grok.loadDefault(function(patterns) {
+                var pattern = patterns.createPattern(self.pattern);
+                var liner = readline.createInterface({
+                    input: stream,
+                    terminal: false
+                });
+
+                liner.on('line', function(line) {
+                    pending++;
+                    sync(function() {
```

Comment:
Yes with `sync` we don't need the pending logic but without it nothing here is synchronous as the node-grok library calls back with the parsed result asynchronously (due node-grok itself using the oniguruma parser which is asynchronous as well, I even went digging into the code myself to see if there were other sync methods, but no such luck: https://github.com/Beh01der/node-grok/blob/master/lib/index.js#L15)  and that was the reason I had to use the pending logic which was super buggy. I'll fix up things by removing the silly pending logic which is no longer in use.